src/layouts/Modal.tsx

```diff
@@ -0,0 +1,140 @@
+import { Fragment, useRef } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import type { FC } from "react";
+import { XMarkIcon } from "@heroicons/react/24/outline";
+import { useTranslation } from "react-i18next";
+
+type Props = {
+  children?: React.ReactNode;
+  // closeable?: boolean;
+  closeFunc: () => void;
+  backFunc?: () => void;
+  submitFunc?: () => void;
+  title: string;
+  submitLabel?: string;
+};
+
+const Modal: FC<Props> = ({
+  // closeable = true,
+  closeFunc,
+  backFunc,
+  children,
+  title,
+  submitLabel,
+  submitFunc,
+}) => {
+  const { t } = useTranslation();
+
+  const cancelButtonRef = useRef(null);
+
+  return (
+    <Transition.Root show as={Fragment}>
+      <Dialog
+        as="div"
+        className="relative z-10"
+        initialFocus={cancelButtonRef}
+        onClose={() => closeFunc()}
+      >
+        <Transition.Child
+          as={Fragment}
+          enter="ease-out duration-300"
+          enterFrom="opacity-0"
+          enterTo="opacity-100"
+          leave="ease-in duration-200"
+          leaveFrom="opacity-100"
+          leaveTo="opacity-0"
+        >
+          <div className="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity" />
+        </Transition.Child>
+
+        <div className="fixed inset-0 z-10 overflow-y-auto">
+          <div className="flex min-h-full items-end justify-center p-4 text-center sm:items-center sm:p-0">
+            <Transition.Child
+              as={Fragment}
+              enter="ease-out duration-300"
+              enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+              enterTo="opacity-100 translate-y-0 sm:scale-100"
+              leave="ease-in duration-200"
+              leaveFrom="opacity-100 translate-y-0 sm:scale-100"
+              leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+            >
+              <Dialog.Panel className="relative transform  overflow-hidden rounded-lg bg-white text-left shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-lg">
+                <form
+                  onSubmit={(event) => {
+                    event.preventDefault();
+                    submitFunc && submitFunc();
+                  }}
+                >
+                  <div className="absolute right-0 top-0 hidden pr-4 pt-4 sm:block">
+                    <button
+                      type="button"
+                      className="rounded-md bg-white dark:bg-zinc-900 text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
```

Comment:
```suggestion
                      className="rounded-md bg-gray-300 dark:bg-zinc-900 text-black dark:text-white  hover:text-gray-500 hover:dark:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
```

The X with white text is more clear in dark mode. Maybe should also have a box in light mode, so it is more consistent.